packages/gestalt/src/Video.js

```diff
@@ -681,32 +724,34 @@ export default class Video extends PureComponent<Props, State> {
       >
         <ColorSchemeProvider id="Video" colorScheme="light">
           <video
-            autoPlay={playing}
-            loop={loop}
-            muted={volume === 0}
-            playsInline={playsInline}
-            poster={poster}
-            preload={preload}
-            src={typeof src === 'string' ? src : undefined}
-            ref={this.setVideoRef}
+            autoPlay={autoplay}
             className={styles.video}
+            {...((crossOrigin ? { crossOrigin } : { ...null }): {|
+              crossOrigin?: CrossOrigin,
+            |})}
             disableRemotePlayback={disableRemotePlayback}
+            loop={loop}
+            muted={volume === 0}
+            {...(objectFit ? { style: { objectFit } } : null)}
```

Comment:
ditto here

Reply:
I can do some research and cleanup. Follow up PR as I dont want to add another potential source of bugs with this change

Reply:
DONE